testkit/src/main/scala/pureconfig/ConfigReaderMatchers.scala

```diff
@@ -26,12 +26,10 @@ trait ConfigReaderMatchers { this: AnyFlatSpec with Matchers =>
   def failWith(failure: ConfigReaderFailure): Matcher[ConfigReader.Result[Any]] =
     be(Left(ConfigReaderFailures(failure)))
 
-  def failWithType[Reason <: FailureReason: ClassTag]: Matcher[ConfigReader.Result[Any]] =
+  def failWithConvertFailureOf[Reason <: FailureReason: ClassTag]: Matcher[ConfigReader.Result[Any]] =
```

Comment:
Do we really have to do this? This is significantly more verbose than before and on this particular instance I don't think we gain much in terms of conciseness.

Reply:
The overload resolution rules change in Scala 3 (https://dotty.epfl.ch/docs/reference/changed-features/overload-resolution.html) so we have an ambiguous overload even with the `DummyImplicit` trick. I suppose we can rename this differently for conciseness (e.g. `failWithReason`?).